src/receive_imf.rs

```diff
@@ -2268,111 +2280,151 @@ async fn apply_group_changes(
         }
     }
 
-    if allow_member_list_changes {
+    // These are for adding info messages about implicit membership changes, so they are only
+    // filled when such messages are needed.
+    let mut added_ids = HashSet::<ContactId>::new();
+    let mut removed_ids = HashSet::<ContactId>::new();
+
+    if let Some(ref chat_group_member_timestamps) = mime_parser.chat_group_member_timestamps() {
+        let expected_timestamps_count = to_ids.len() + past_ids.len();
+        if chat_group_member_timestamps.len() == expected_timestamps_count {
+            context
+                .sql
+                .transaction(|transaction| {
+                    for (contact_id, ts) in iter::zip(
+                        to_ids.iter(),
+                        chat_group_member_timestamps.iter().take(to_ids.len()),
+                    ) {
+                        if *contact_id != from_id {
+                            let modified = transaction.execute(
```

Comment:
Performance: Wouldn't it be more performant to `prepare(…)` the statement once and then call `statement.execute(params)` in the loop? I didn't measure, though.

Same for the `for` loop below and in `create_group(…)`, `add_to_chat_contacts_table(…)`, and `update_chat_contacts_table(…)`